americangut/taxtree.py

```diff
@@ -123,6 +153,28 @@ def build_tree_from_taxontable(table):
     return tree, sample_taxa_lookup
 
 
+def build_persample_tree_from_taxontable(table):
+    """Construct per-sample trees from a taxon table
+
+    yields (sample_id, tree). Each node in the tree (tip and nontip) include
+    the taxon name, associated sequence count as well as the relative
+    abundance.
+    """
+    samp_ids = table.ids()
+    obs_ids_tmp = table.ids(axis='observation')
+    obs_ids = [[t.strip() for t in i.split(';')] for i in obs_ids_tmp]
```

Comment:
Maybe make a note in the docstring that this is the expected format for the taxonomy metadata? biom supports multiple formats, iirc